File: benchmarks/java/B7Concurrent.java
import java.util.concurrent.atomic.AtomicLong;
public class B7Concurrent {
    public static void main(String[] args) throws Exception {
        AtomicLong counter = new AtomicLong(0);
        Thread[] threads = new Thread[8];
        for (int i = 0; i < 8; i++) {
            threads[i] = new Thread(() -> {
                for (int j = 0; j < 10000000; j++) counter.incrementAndGet();
            });
            threads[i].start();
        }
        for (Thread t : threads) t.join();
        System.out.println(counter.get());
    }
}

File: benchmarks/kotlin/b7_concurrent.kt
import java.util.concurrent.atomic.AtomicLong
fun main() {
    val counter = AtomicLong(0)
    val threads = Array(8) {
        Thread { repeat(10_000_000) { counter.incrementAndGet() } }.also { it.start() }
    }
    threads.forEach { it.join() }
    println(counter.get())
}
